RUFAS/biophysical/animal/nutrients/nutrition_supply_calculator.py

```diff
@@ -0,0 +1,651 @@
+                    0.703 * actual_metabolizable_energy[feed.info.rufas_id]
+                    - 0.19
+                    + ((0.097 * actual_metabolizable_energy[feed.info.rufas_id] + 0.19) / 97) * (feed.info.EE - 3.0)
+                )
+            else:
+                energy = 0.703 * actual_metabolizable_energy[feed.info.rufas_id] - 0.19
+            actual_lactation_net_energy[feed.info.rufas_id] = energy
+        total = sum([feed.amount * actual_lactation_net_energy[feed.info.rufas_id] for feed in feeds])
+
+        return total
+
+    @classmethod
+    def _calculate_actual_growth_net_energy(
+        cls, feeds: list[FeedInRation], actual_metabolizable_energy: dict[RUFAS_ID, float]
+    ) -> float:
+        """
+        Calculates actual net energy for growth of feeds.
+
+        Parameters
+        ----------
+        feeds : list[FeedInRation]
+            List of feeds in ration, including the amount and nutritive properties.
+        actual_metabolizable : dict[RUFAS_ID, float]
+            Mapping of RuFaS Feed IDs to the actual metabolizable energy content (Mcal) of the feed.
+
+        Returns
+        -------
+        float
+            Total actual net energy available for growth in the ration (Mcal).
+
+        References
+        ----------
+        Animal Scientific Documentation [A.Cow.E.9]-[A.Heifer.E.9]
+
+        """
+        actual_growth_net_energy: dict[RUFAS_ID, float] = {}
+
+        for feed in feeds:
+            if feed.info.feed_type is FeedComponentType.MINERAL:
+                energy = 0.0
+            elif feed.info.is_fat is True:
+                energy = 0.55 * actual_metabolizable_energy[feed.info.rufas_id]
+            else:
+                energy = (
+                    1.42 * actual_metabolizable_energy[feed.info.rufas_id]
+                    - 0.174 * actual_metabolizable_energy[feed.info.rufas_id] ** 2
+                    + 0.0122 * actual_metabolizable_energy[feed.info.rufas_id] ** 3
+                    - 1.65
+                )
+            actual_growth_net_energy[feed.info.rufas_id] = energy
+        total = sum([feed.amount * actual_growth_net_energy[feed.info.rufas_id] for feed in feeds])
+
+        return total
+
+    @classmethod
+    def _calculate_calcium_supply(cls, feeds: list[FeedInRation]) -> float:
+        """
+        Calculates the calcium supply in the ration.
+
+        Parameters
+        ----------
+        feeds : list[FeedInRation]
+            List of feeds in ration, including the amount and nutritive properties.
+
+        Returns
+        -------
+        float
+            Total digestible calcium supply in the ration (g).
+
+        References
+        ----------
+        Animal Scientific Documentation [A.Cow.E.16]-[A.Heifer.E.16]
+
+        """
+        calcium_digestibility: dict[RUFAS_ID, float] = {}
+
+        for feed in feeds:
+            if feed.info.feed_type is FeedComponentType.FORAGE:
+                ca_digestibility = 0.3
```

Comment:
```suggestion
                digestibility = 0.3
```
minor suggestion to match the energy ones which were more general within the for loops.